packages/govuk-frontend/.browserslistrc

```diff
@@ -1,13 +1,25 @@
 # This list builds on the GOV.UK service manual's browser testing recommendations
 # https://www.gov.uk/service-manual/technology/designing-for-different-browsers-and-devices
 
-> 0.1%
-last 2 Chrome versions
-last 2 Firefox versions
-last 2 Edge versions
+[javascripts]
+> 0.1% in GB and supports es6-module
+last 6 Chrome versions
+last 6 Firefox versions
+last 6 Edge versions
 last 2 Samsung versions
-Safari >= 9
-iOS >= 9
+Firefox ESR
+Safari >= 11 and not Safari < 11
+iOS >= 11 and not iOS < 11
```

Comment:
Based on [the decisions we made on Tuesday](https://github.com/alphagov/govuk-design-system-architecture/pull/25), I'd expect this to just be:

```browserslist
[javascripts]
supports es6-module
```

By restricting it to only browsers with `>0.1% marketshare` we're back in a place where we may ship JS that is not parseable in all browsers that support `type="module"`?

Additionally, is there benefit in listing out specific browsers when we know they'll be covered by `supports es6-module` anyway?

Reply:
@36degrees Of course! I'm torn on wanting to let browsers go from Grade A, to Grade B, to Grade C

Using the query `> 0.1% in GB and supports es6-module` lets us gracefully (in future) go from 100% polyfill coverage, to letting older browsers slowly fail feature detection and drop down a grade

Let me know if you'd rather have a fixed point in time instead—I'll fit in with you

(Adding specific browsers was to prevent the sliding scale from going too far)

Reply:
Thanks for the catch up with @36degrees

For safety let's go with `supports es6-module`

Should we find that Babel is adding some weighty transforms/plugins that we could instead guard with "feature detection", we'll exclude them in the Babel config instead

Reply:
Most interesting side effect of this change is over in:

* https://github.com/alphagov/govuk-frontend/pull/3820

We're now seeing all `let` and `const` transformed back to `var` for `Safari < 11` 😮 

Odd, but let's dig further

Babel reports that the transform was applied by [`@babel/plugin-transform-block-scoping`](https://babeljs.io/docs/babel-plugin-transform-block-scoping) to fix [slowdowns in Safari](https://bugs.webkit.org/show_bug.cgi?id=199866). Appears that we set [`safari10: true` in Terser](https://github.com/alphagov/govuk-frontend/blob/d7a56ccde89de612a90e1ee5b232f35c924e8150/packages/govuk-frontend/rollup.release.config.mjs#L37) for the same reason

You might find ESBuild's explanation is easier to follow than the [WebKit bug](https://bugs.webkit.org/show_bug.cgi?id=199866)

>### [Top-level `var`](https://esbuild.github.io/faq/#top-level-var)
>In JavaScript, `let`, `const`, and `class` declarations all introduce TDZ checks while var declarations do not. Since bundling typically merges many modules into a single very large top-level scope, the performance impact of these TDZ checks can be pretty severe. Converting top-level `let`, `const`, and `class` declarations into `var` helps automatically make your code faster.

Reply:
When adding `debug: true` to the Babel `@babel/preset-env` config you'll see the following:

```console
Using plugins:
  transform-unicode-sets-regex { chrome < 112, edge < 112, firefox, ios, opera < 98, safari, samsung }
  proposal-class-static-block { chrome < 94, edge < 94, firefox < 93, ios, opera < 80, safari, samsung < 17 }
  proposal-private-property-in-object { chrome < 91, edge < 91, firefox < 90, ios < 15, opera < 77, safari < 15, samsung < 16 }
  proposal-class-properties { chrome < 74, edge < 79, firefox < 90, ios < 15, opera < 62, safari < 14.1, samsung < 11 }
  proposal-private-methods { chrome < 84, edge < 84, firefox < 90, ios < 15, opera < 70, safari < 15, samsung < 14 }
  proposal-numeric-separator { chrome < 75, edge < 79, firefox < 70, ios < 13, opera < 62, safari < 13, samsung < 11 }
  proposal-logical-assignment-operators { chrome < 85, edge < 85, firefox < 79, ios < 14, opera < 71, safari < 14, samsung < 14 }
  proposal-nullish-coalescing-operator { chrome < 80, edge < 80, firefox < 72, ios < 13.4, opera < 67, safari < 13.1, samsung < 13 }
  proposal-optional-chaining { chrome < 91, edge < 91, firefox < 74, ios < 13.4, opera < 77, safari < 13.1, samsung < 16 }
  proposal-json-strings { chrome < 66, edge < 79, firefox < 62, ios < 12, opera < 53, safari < 12, samsung < 9 }
  proposal-optional-catch-binding { chrome < 66, edge < 79, ios < 11.3, opera < 53, safari < 11.1, samsung < 9 }
  transform-parameters { edge < 18, ios, safari }
  proposal-async-generator-functions { chrome < 63, edge < 79, ios < 12, opera < 50, safari < 12 }
  proposal-object-rest-spread { edge < 79, ios < 11.3, safari < 11.1 }
  transform-dotall-regex { chrome < 62, edge < 79, firefox < 78, ios < 11.3, opera < 49, safari < 11.1 }
  proposal-unicode-property-regex { chrome < 64, edge < 79, firefox < 78, ios < 11.3, opera < 51, safari < 11.1, samsung < 9 }
  transform-named-capturing-groups-regex { chrome < 64, edge < 79, firefox < 78, ios < 11.3, opera < 51, safari < 11.1, samsung < 9 }
  transform-async-to-generator { ios < 11, safari < 11 }
  transform-template-literals { ios < 13, safari < 13 }
  transform-function-name { edge < 79 }
  transform-unicode-regex { ios < 12, safari < 12 }
  transform-block-scoping { ios < 11, safari < 11 }
  proposal-export-namespace-from { chrome < 72, edge < 79, firefox < 80, ios, opera < 60, safari, samsung < 11.0 }
  syntax-dynamic-import
  syntax-top-level-await
  syntax-import-meta
```

Reply:
Interesting! Any thoughts on what we should do?

I guess we might want to do some performance testing on the affected Safari versions to see what the impact would be of disabling that transformation?

Reply:
Let's keep the config for now and assume (future) Babel will save our bacon many more times 😓

But ideally let's see if Rollup avoids the issue already, since those Babel transforms might be 100% unnecessary

This PR is good to go though